openam-shared/src/main/java/org/forgerock/openam/shared/security/crypto/AESWrapEncryption.java

```diff
@@ -66,7 +67,11 @@ public class AESWrapEncryption implements AMEncryption, ConfigurableKey {
     private static final int AESWRAP_BLOCK_SIZE = 8;
 
     private static final int CACHE_SIZE = Integer.getInteger("amCryptoCacheSize", 500);
-    private static final CipherProvider CIPHER_PROVIDER = Providers.cipherProvider("AESWrap", null, CACHE_SIZE);
+
+    private static final String CIPHER_PROVIDER_ALGORITHM =
```

Comment:
This feels like a hacky solution that deserves XXX comment.

Proper fix would be increasing block size to 16 bytes, however that would require changing unit tests. Such change feels unnecessary considering that with Java 17 the KW algorithm (`AES128_KW_PKCS5Padding`) is part of the JRE and this whole class can be dropped. So this class can be considered as deprecated in a sense.

Reply:
Just checked again and my comment is incorrect.